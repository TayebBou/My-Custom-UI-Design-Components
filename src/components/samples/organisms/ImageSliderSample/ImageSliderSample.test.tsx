import { render } from "@testing-library/react";
import ImageSliderSample from "./ImageSliderSample";

describe("ImageSliderSample component", () => {
  test("should render correctly", () => {
    // Arrange
    const { container } = render(<ImageSliderSample />);
    // Assert
    expect(container).toMatchSnapshot();
  });
});
