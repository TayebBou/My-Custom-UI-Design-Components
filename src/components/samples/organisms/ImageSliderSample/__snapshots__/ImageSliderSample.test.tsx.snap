// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSliderSample component should render correctly 1`] = `
<div>
  <h2>
    ImageSlider Component
  </h2>
  <div
    class="wrapper"
  >
    <div
      class="fullscreen"
    >
      <div
        class="full-images"
        data-testid="bigImagesDiv"
      >
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_1.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_2.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_3.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_4.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_5.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_6.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_7.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_8.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_9.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_10.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <img
          alt="full-size"
          src="/images/MERCEDES_CLASSE_C_11.webp"
          style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
        />
        <button
          class="button slider-button right"
          type="button"
        >
          &gt;
        </button>
        <button
          class="button slider-button left"
          disabled=""
          type="button"
        >
          &lt;
        </button>
        <span
          class="span"
        >
          1
           / 
          11
        </span>
        <button
          class="button button "
          type="button"
        >
          <img
            alt="full screen"
            class="img"
            src="full_screen.png"
            style="filter: none;"
          />
        </button>
      </div>
      <div
        class="mini-images"
        data-testid="imagesDiv"
      >
        <button
          class="button slider-button"
          disabled=""
          type="button"
        >
          &lt;
        </button>
        <div
          class="images"
          data-testid="miniImagesDiv"
        >
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_1.webp"
            style="max-width: 92px; margin: 1.5em 0px; outline: solid 0.2em white; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_2.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_3.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_4.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_5.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_6.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_7.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_8.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_9.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_10.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
          <img
            alt="thumbnail"
            src="/images/MERCEDES_CLASSE_C_11.webp"
            style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px); cursor: pointer;"
          />
        </div>
        <button
          class="button slider-button"
          type="button"
        >
          &gt;
        </button>
      </div>
    </div>
  </div>
  <div>
    <h3>
      Props Documentation
    </h3>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th
            class="header"
          >
            Prop Name
          </th>
          <th
            class="header"
          >
            Type
          </th>
          <th
            class="header"
          >
            Description
          </th>
          <th
            class="header"
          >
            Modify
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="cell"
          >
            photos
          </td>
          <td
            class="cell"
          >
            IPhoto[]
          </td>
          <td
            class="cell"
          >
            An array of photo objects. Each object must contain a 'photos' array, and each photo inside it should have properties like 'url_full'.
          </td>
          <td
            class="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
